src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1366,7 +1366,7 @@ public static bool IsPartial(this TypeSymbol type)
 
         public static bool HasFileLocalTypes(this TypeSymbol type)
         {
-            var foundType = type.VisitType(predicate: (type, _, _) => type is SourceMemberContainerTypeSymbol { IsFileLocal: true }, arg: (object?)null);
+            var foundType = type.VisitType(predicate: (type, _, _) => type is NamedTypeSymbol { IsFileLocal: true }, arg: (object?)null);
```

Comment:
>IsFileLocal [](http://example.com/codeflow?start=96&length=11)

I see several more places where access to ```IsFileLocal``` property is "surrounded" with a type check. For example:
```
        private static BestSymbolLocation GetLocation(CSharpCompilation compilation, Symbol symbol)
        {
            if (symbol is SourceMemberContainerTypeSymbol { IsFileLocal: true })
```

Consider getting rid of the type checks, unless they are significant. Or adding a comment (about the significance) otherwise.